apps/cowswap-frontend/src/modules/tradeFormValidation/hooks/useTradeFormButtonContext.ts

```diff
@@ -1,14 +1,42 @@
 import { useMemo } from 'react'
 
+import { isAddress } from '@cowprotocol/common-utils'
 import { useWalletDetails } from '@cowprotocol/wallet'
 
 import { useToggleWalletModal } from 'legacy/state/application/hooks'
 
 import { useInjectedWidgetParams } from 'modules/injectedWidget'
-import { useAmountsToSign, useDerivedTradeState, useWrapNativeFlow } from 'modules/trade'
+import { TradeDerivedState, useAmountsToSign, useDerivedTradeState, useIsCurrentTradeBridging, useWrapNativeFlow } from 'modules/trade'
 import { useTradeQuote } from 'modules/tradeQuote'
 
-import { TradeFormButtonContext } from '../types'
+import { SUPPORTED_ENS_CHAINS } from '../services/validateTradeForm'
+import { RecipientValidationResult, TradeFormButtonContext } from '../types'
+
+const RECIPIENT_VALIDATION_MESSAGES = {
+  ENS_NOT_SUPPORTED: 'Please use a blockchain address. ENS names are not supported on this chain.',
+  INVALID_RECIPIENT: 'Enter a valid blockchain address or ENS name.',
+  INVALID_ADDRESS: 'Enter a valid blockchain address.',
+} as const
+
+function createRecipientValidation(derivedState: TradeDerivedState, isBridging: boolean): RecipientValidationResult {
```

Comment:
`context.recipientValidation` is only used in `tradeButtonsMap.tsx`.
The context already includes `derivedState`, `isBridging` can be calculated using this state.
It looks like the validation message can and should be calculated in `tradeButtonsMap.tsx` in case of `TradeFormValidation.RecipientInvalid`. So, no need to define this function in `useTradeFormButtonContext.ts`.
Also, some of the checks are duplicated (chainSupportsENS, isENSName).